libgrust/libformat_parser/src/lib.rs

```diff
@@ -327,16 +327,20 @@ mod ffi {
 
 // FIXME: Rename?
 pub mod rust {
-    use generic_format_parser::{ParseMode, Parser, Piece};
-
+    use crate::ffi::ParseMode;
+    use generic_format_parser::{Parser, Piece};
     pub fn collect_pieces(
         input: &str,
         style: Option<usize>,
         snippet: Option<String>,
         append_newline: bool,
+        parse_mode: ParseMode,
     ) -> Vec<Piece<'_>> {
-        let parser = Parser::new(input, style, snippet, append_newline, ParseMode::Format);
-
+        let converted_parse_mode = match parse_mode {
+            ParseMode::Format => generic_format_parser::ParseMode::Format,
+            ParseMode::InlineAsm => generic_format_parser::ParseMode::InlineAsm,
```

Comment:
this is good, if you want one thing you could do to avoid the nameclash is the following:

```
use generic_format_parser::ParseMode as RustcParseMode;
```

and then use it like so:


```suggestion
        let converted_parse_mode = match parse_mode {
            ParseMode::Format => RustcParseMode::Format,
            ParseMode::InlineAsm => RustcParseMode::InlineAsm,
```

or we could also add an implementation for `Into<RustcParseMode> for ParseMode`.

but this is definitely just a nit/suggestion and not a requirement

Reply:
:) thank you for reviewing this